node/pegnet/txrelation.go

```diff
@@ -50,3 +50,19 @@ func InsertTransactionRelation(tx *sql.Tx, adrID int64, entryHash factom.Bytes32
 	}
 	return lastID, nil
 }
+
+// IsReplayTransaction returns true if there exist any transaction relations in the
+// "pn_address_transactions" table.
+func (p *Pegnet) IsReplayTransaction(tx *sql.Tx, entryHash *factom.Bytes32) (bool, error) {
+	rows, err := tx.Query(`SELECT * FROM "pn_address_transactions" WHERE "entry_hash" = ?;`)
+	if err != nil {
+		return false, err
+	}
+	defer rows.Close()
+	err = rows.Err()
+	if err != nil {
+		return false, err
+	}
+	// If there is any result, then we know the transaction has been executed before and thus a replay.
+	return rows.Next(), nil
```

Comment:
We should check for `err == sql.ErrNoRows`, as that means there is no replay.

Reply:
You can also use `QueryRow` if you only expect 1 row, and then you don't need to close the rows

Reply:
If using `QueryRow` then we'd need to also call `Scan()` in order to see any errors.

I think `Query` will not return an `sql.ErrNoRows` if the result set is empty, but I just added the check anyways